ibis/expr/types/structs.py

```diff
@@ -75,6 +75,11 @@ def struct(
     """
     import ibis.expr.operations as ops
 
+    if isinstance(value, StructValue):
+        if type is not None:
+            return value.cast(type)
+        return value
```

Comment:
This functionality seems unrelated to the original issue and needs tests. Can you add this feature in a follow up PR?

Reply:
Oops this was originally needed but then I adjusted the implementation and never reverted this. Will do. I think it's a good change though.

Reply:
moved to #8666